frame/lending/src/lib.rs

```diff
@@ -1066,58 +1231,20 @@ pub mod pallet {
 
 			ensure!(
 				!matches!(
-					T::Vault::available_funds(&market.borrow, market_account)?,
+					T::Vault::available_funds(&market.borrow_asset_vault, market_account)?,
 					FundsAvailability::MustLiquidate
 				),
 				Error::<T>::MarketIsClosing
 			);
 
 			Ok(())
 		}
-
-		fn can_repay_borrow(
-			market_id: &MarketIndex,
-			from: &T::AccountId,
-			beneficiary: &T::AccountId,
-			repay_amount: BorrowAmountOf<Self>,
-			owed: BorrowAmountOf<Self>,
-			borrow_asset_id: <T as DeFiComposableConfig>::MayBeAssetId,
-			market_account: &T::AccountId,
-		) -> Result<(), DispatchError> {
-			let latest_borrow_timestamp = BorrowTimestamp::<T>::get(market_id, beneficiary);
-			ensure!(latest_borrow_timestamp.is_some(), Error::<T>::BorrowDoesNotExist);
-			if let Some(timestamp) = latest_borrow_timestamp {
-				ensure!(
-					timestamp != Self::last_block_timestamp(),
-					Error::<T>::BorrowAndRepayInSameBlockIsNotSupported
-				);
-			}
-			ensure!(
-				repay_amount > <Self as DeFiEngine>::Balance::zero(),
-				Error::<T>::RepayAmountMustBeGraterThanZero
-			);
-			ensure!(repay_amount <= owed, Error::<T>::CannotRepayMoreThanBorrowAmount);
-			ensure!(
-				<T as Config>::MultiCurrency::can_withdraw(borrow_asset_id, from, repay_amount)
-					.into_result()
-					.is_ok(),
-				Error::<T>::CannotWithdrawFromProvidedBorrowAccount
-			);
-			ensure!(
-				<T as Config>::MultiCurrency::can_deposit(
-					borrow_asset_id,
-					market_account,
-					repay_amount
-				)
-				.into_result()
-				.is_ok(),
-				Error::<T>::TransferFailed
-			);
-
-			Ok(())
-		}
 	}
 
+	/////////////////////////////////////////////////////////////////////////
+	// DefiEngine implementation
```

Comment:
DefiEngine implementation 

is

impl DeFiEngine for Pallet. 

so you type same 2 times, not sure why. 

just make impl more visual